app/models/mixins/miq_policy_mixin.rb

```diff
@@ -22,21 +26,14 @@ def remove_policy(policy)
     reload
   end
 
-  def get_policies(mode = nil)
-    ns = "/miq_policy"
-    cat = "assignment"
-    tag_list(:ns => ns, :cat => cat).split.collect do |t|
+  def get_policies
+    policy_tags.collect do |t|
       klass, id = t.split("/")
       next unless ["miq_policy", "miq_policy_set"].include?(klass)
-      policy = klass.camelize.constantize.find_by(:id => id.to_i)
-      policy if mode.nil? || policy.mode == mode
+      klass.camelize.constantize.find_by(:id => id.to_i)
     end.compact
```

Comment:
This is an N+1...I think we can rewrite this to be `O(class_count) == O(1)`  (i.e. make one query for each class type, with all ids queried for that class type)

```ruby
policy_tags
  .map { |t| t.split("/", 2) }
  .group_by(&:first)
  .select { |klass, _ids| ["miq_policy", "miq_policy_set"].include?(klass) }
  .flat_map { |klass, ids| klass.camelize.constantize.where(:id => ids).to_a }
```

Reply:
Not necessarily for this PR.

Reply:
heh - I have a local comment in my code to do this.

@Fryguy Should I specifically do this in another PR? So the changes are distinct.
Or should I throw that in for good measure?

Reply:
I think you should add it here because the goal of this effort is to provide `has_policies?` virtual attr in the most efficient way we can.